src/mxarray.jl

```diff
@@ -686,11 +686,10 @@ end
 
 # deep conversion from MATLAB variable to Julia array
 
-jvalue(mx::MxArray, ty::Type{Array})  = jarray(mx)
-jvalue(mx::MxArray, ty::Type{Vector}) = jvector(mx)
-jvalue(mx::MxArray, ty::Type{Matrix}) = jmatrix(mx)
-jvalue(mx::MxArray, ty::Type{Number}) = jscalar(mx)::Number
-jvalue(mx::MxArray, ty::Type{String}) = jstring(mx)::String
-jvalue(mx::MxArray, ty::Type{Dict}) = jdict(mx)
-jvalue(mx::MxArray, ty::Type{SparseMatrixCSC}) = jsparse(mx)
-
+convert(::Type{Array}, mx::MxArray)  = jarray(mx)
+convert(::Type{Vector}, mx::MxArray) = jvector(mx)
+convert(::Type{Matrix}, mx::MxArray) = jmatrix(mx)
+convert(::Type{Number}, mx::MxArray) = jscalar(mx)
```

Comment:
Probably best to leave the type assert here although I'm not sure it helps much.